### ISSUE TITLE
Move (or copy) multiple events to another speaker

### ISSUE THREAD

**Issue by sarkipo** [EXMARaLDA/exmaralda#473]
Similar to Copy/Paste multiple events #471 
related to #467 

Particularly useful when a portion of speech has been wrongly transcribed as speaker A but is actually from speaker B.
In a simple case, move each event to the corresponding tier (category) and TLI of speaker B (so no need to ask for destination tier).

**Comment by berndmoos** [comment]
@sarkipo : could you give me an idea if a sequence of
- copy structure
- remove events
- paste structure
does the job in a way you could live with? It seems like an unnecessary duplication of functionality if I make this an extra item. If there's something beyond copy/remove/paste that I don't see please let me know.

**Comment by sarkipo** [comment]
In principle, yes

In practice, as of now, it can be not very handy since I must select all the tiers. Ergo they must all be unhidden, and with our 20+ tiers pro speaker they do not always fit on the screen :-)
(I've hidden some of them here:)
![grafik](https://github.com/Exmaralda-Org/exmaralda/assets/11090777/2c40939e-943f-47f7-8db0-77e454ae9364)

Now, there is a problem I had seen before: If there are events spanning multiple TLIs above and below, but single TLIs in the middle (and this is our standard layout), I cannot properly make a rectangular selection: only the first event gets selected in each tier.
![grafik](https://github.com/Exmaralda-Org/exmaralda/assets/11090777/22698cfa-aab0-45ea-8ade-3df304f795e5)

**Comment by sarkipo** [comment]
It could be made easier if there were a command "Select all dependent annotations" (given a selection in a transcription tier, select all dependent annotations in all tiers). Then I could select just the words in tx, use that command to get the entire selection I need, copy structure, remove events, paste structure.
(Especially if "Paste structure" would also work with just the transcription tier selected...)

**Comment by berndmoos** [comment]
Yes, I see. The kind of selection you mention is indeed not possible, but I am afraid that is something deep down in the GUI component (not mine) that won't be easy to change. I need to think about how INEL specific or not the solutions you suggest would be. In any case, this is not the most straightforward issue.

**Comment by berndmoos** [comment]
Make a menu item "Move block to other speaker"?

**Comment by berndmoos** [comment]
And functionality to "select block"

**Comment by berndmoos** [comment]
(1) Click in suitable event:

![Image](https://github.com/user-attachments/assets/704ccb66-39f8-447b-a4f4-232a178a533a)

(2) Edit > Select block (or Ctrl + B, or context menu)

![Image](https://github.com/user-attachments/assets/4c00cb8d-520a-4c78-bd58-c0208bb40048)

(3) Event > Move to other speaker... (or context menu)

![Image](https://github.com/user-attachments/assets/3760acef-2942-4093-bc58-deb368aed14f)

(4) Select target speaker

![Image](https://github.com/user-attachments/assets/12cedba2-275b-4060-ab80-cfd223107b1b)

(5) TADA!

![Image](https://github.com/user-attachments/assets/78db8017-1deb-4b83-bb37-ea5efabdbe36)

This involves a lot of checks, most importantly if an empty target structure identical to the source structure exists.

**Comment by sarkipo** [comment]
Aah, that's working so nicely... I wish we had it when we worked on Selkup...

One tricky file, though: I've tried a file with 5 speakers, and they seem to be divided in two groups: I can only move between KEP and KTP and between KuAI, IF and KuMP, but not across those groups.

[KuMP_IF_KTP_KEP_196X_Plans_conv.zip](https://github.com/user-attachments/files/20573794/KuMP_IF_KTP_KEP_196X_Plans_conv.zip)

**Comment by berndmoos** [comment]
> One tricky file, though

Argh, the size of that Partitur!!!!!

@sarkipo: I checked for speaker KEP. It seems he has a tier "ExLocPoss" which, for example, speaker KuMP does not have. So the structures don't match, and the tier bundle cannot be moved. I could check for other speakers, but my head's spinning enough as it is...
